repositories/post_repository.py

```
@@ -0,0 +1,65 @@
+from common.config import settings
+
+
+class PostRepository:
+    def __init__(self) -> None:
+        self.file_path = settings.posts_json_path
+        if not os.path.exists(self.file_path):
+            self._save_all([])
+
+    # 내부용 함수
+    def _load_all(self) -> list[dict]:
+        with open(self.file_path, "r", encoding="utf-8") as f:
```

Ruff (0.14.11)

Unnecessary mode argument

Remove mode argument

(UP015)